src/main/scala/io/iohk/ethereum/mpt/MerklePatriciaTrie.scala

```diff
@@ -241,37 +236,37 @@ class MerklePatriciaTrie[K, V] private (private val rootHash: Option[Array[Byte]
   }
 
   @tailrec
-  private def get(node: Node, searchKey: Array[Byte]): Option[Array[Byte]] = node match {
-    case LeafNode(key, value, _) =>
-      if (key sameElements searchKey) Some(value) else None
-    case extNode@ExtensionNode(sharedKey, _, _) =>
+  private def get(node: MptNode, searchKey: Array[Byte]): Option[Array[Byte]] = node match {
```

Comment:
shouldn't it be better to change `searchKey` to ByteArray and the result to `Option[ByteString]` ?